samples/subsys/shell/shell_module/README.rst

```diff
@@ -0,0 +1,200 @@
+      shell_set_bypass(sh, NULL);
+   }
+
+* Set a Login command
+
+.. code-block:: c
+
+   static int cmd_login(const struct shell *sh, size_t argc, char **argv)
+   {
+      /* Perform login verification */
+      ...
+
+      /* Remove root command to make other commands accessible */
+      shell_set_root_cmd(NULL);
+
+      /* Login clean-up */
+      ...
+
+      return 0;
+   }
+
+
+   SHELL_COND_CMD_ARG_REGISTER(CONFIG_SHELL_START_OBSCURED, login, NULL,
+               "<password>", cmd_login, 2, 0);
+
+   void init_func(void)
+   {
+      ...
+
+      /* Set Root command, which verifies login information */
+      shell_set_root_cmd("login");
+
+      ...
+   }
+
+* Obscure user-input with asterisks
+
+.. code-block:: c
+
+      /* Enable obscure mode */
+      shell_obscure_set(sh, true);
+
+      /* User-input is replaced with asterisks */
+      ...
+
+      /* Disable obscure mode */
+      shell_oscure_set(sh, false);
+      ...
+
+Requirements
+************
+
+* A target configured with the shell interface, exposed through any of its backends.
+
+Building and Running
+********************
+
+This sample can be found under :zephyr_file:`samples/subsys/shell/shell_module`
+in the Zephyr tree.
+
+The sample can be built for several platforms.
+
+Emulation Targets
+=================
+
+The sample may run on emulation targets. The following commands build the
+application for the qemu_x86.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/subsys/shell/shell_module
+   :host-os: unix
+   :board: qemu_x86
+   :goals: run
+   :compact:
+
+After running the application, the console displays the shell interface, and
+shows the shell prompt, at which point the user may start the interaction.
+
+On-Hardware
+===========
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/subsys/shell/shell_module
+   :host-os: unix
+   :board: nrf52840dk/nrf52840
+   :goals: flash
+   :compact:
+
+Sample Output
+*************
+
+.. code-block:: console
+
+    uart:~$
+      bypass              clear               date
+      demo                device              devmem
+      dynamic             help                history
+      kernel              log                 log_test
+      rem                 resize              retval
+      section_cmd         shell               shell_uart_release
+      stats               version
+    uart:~$ kernel
+    kernel - Kernel commands
+    Subcommands:
+      cycles     : Kernel cycles.
+      stacks     : List threads stack usage.
+      threads    : List kernel threads.
+      uptime     : Kernel uptime. Can be called with the -p or --pretty options
+      version    : Kernel version.
+      sleep      : ms
+      log-level  : <module name> <severity (0-4)>
+    uart:~$ kernel version
+    Zephyr version 3.6.99
+    uart:~$
+
+Details on Shell Subsystem
+==========================
+
+For more details on the Shell subsystem, check the general Shell documentation:
+:zephyr_file:`doc/services/shell/index.rst`
```

Comment:
does linking to the rst section work from samples? I don't remember.
It's either sample -> general doc or general doc -> sample linking that doesn't.

Reply:
I've changed it to **:ref:`Shell documentation <shell_api>`** and it does appear to work on a local build.